host/xtest/pkcs11_1000.c

```diff
@@ -146,6 +151,92 @@ static void xtest_tee_test_1001(ADBG_Case_t *c)
 	}
 
 	Do_ADBG_EndSubCase(c, "Test C_Get{Slot|Token}Info()");
+	Do_ADBG_BeginSubCase(c, "Test C_GetMechanism{List|Info}()");
+
+	for (i = 0; i < slot_count; i++) {
+		CK_SLOT_ID slot = slot_ids[i];
+		size_t j = 0;
+
+		rv = C_GetMechanismList(slot, NULL, &mecha_count);
+		if (!ADBG_EXPECT_CK_OK(c, rv))
+			goto out;
+
+		if (mecha_count == 0) {
+			if (level)
+				Do_ADBG_Log("Token #%zu: no mechanism", i);
+
+			continue;
+		}
+
+		free(mecha_types);
+		mecha_types = calloc(mecha_count, sizeof(*mecha_types));
+		if (!ADBG_EXPECT_NOT_NULL(c, mecha_types))
+			goto out;
+
+		/* Test specific case: valid buffer reference with 0 count */
+		mecha_count = 0;
+		rv = C_GetMechanismList(slot, mecha_types, &mecha_count);
+		if (!ADBG_EXPECT_CK_RESULT(c, CKR_BUFFER_TOO_SMALL, rv))
+			goto out;
+
+		rv = C_GetMechanismList(slot, mecha_types, &mecha_count);
+		if (!ADBG_EXPECT_CK_OK(c, rv))
+			goto out;
+
+		if (level)
+			Do_ADBG_Log("Token #%zu mechanism capabilities:", i);
+
+		for (j = 0; j < mecha_count; j++) {
+			size_t pos = 0;
+			size_t k = 0;
+			/* 1024byte should be enough, if not truncates  */
+			char log[1024] = { 0 };
```

Comment:
The name `log` may conflict with the standard function `log()`.

Reply:
ok, will change. thanks.